Sage/carbon
# refactor(form): convert to typescript

Convert Form component to typescript. Also extract stories to new `.stories.tsx` file and import into MDX docs.

### Proposed behaviour

Form should be a TypeScript component



### Current behaviour

Form is currently a Javascript component.



### Checklist




#### QA


### Additional context



### Testing instructions



As well as regression testing Form, please check the storybook docs, including all examples, and make sure that nothing (other than code reformatting) has changed in how the examples are presented.
